src/common/evaluable.ts

```diff
@@ -60,10 +60,17 @@ export interface Evaluable {
    * Simplifies this Evaluable when possible.
    *
    * @param {Context} ctx context for the evaluation
-   * @param {string[]} ignoreKeys keys to be considered present even if their not present in the context
+   * @param {false | string[]} alwaysEvaluate keys to be considered present even if they are not present in the context
+   * @param {false | string[]} deferEvaluate keys to be considered not present unless they are also in `alwaysEvaluate`;
+   *  when `alwaysEvaluate` is set to `false` and `deferEvaluate` is an array, every key that is not in `deferEvaluate`
+   *  is considered to be present and thus will be evaluated
    * @returns {Result | Evaluable} simplified value or itself
    */
-  simplify(ctx: Context, ignoreKeys: string[]): Result | Evaluable
+  simplify(
+    ctx: Context,
+    alwaysEvaluate?: false | string[],
+    deferEvaluate?: false | string[]
```

Comment:
Thinking about naming.. would `strictKeys?: string[]` and `skipKeys?: string[]` be better?

Reply:
+1 `strict | optional`

Reply:
Done a292b90d26f2c82ea76701a5aacdc81f7e2a6674